syncstorage/db.go

```diff
@@ -217,23 +219,22 @@ func NewDB(path string, conf *Config) (*DB, error) {
   work is handled by private functions.
 */
 
-// LastModified gets the database modified time
-func (d *DB) LastModified() (modified int, err error) {
+// LastModified returns the top level last modified timestamp
+func (d *DB) LastModified() (int, error) {
 	d.Lock()
 	defer d.Unlock()
 
-	var m sql.NullInt64
+	lastMod, err := getKey(d.db, STORAGE_LAST_MODIFIED)
+	if lastMod == "" || err != nil {
+		return 0, err
```

Comment:
Also worth noting is that the local change I had applied which caused this issue is no longer going to be checked in - it was to fix an edge-case to work around a delete followed by a get returning a 404 instead of a 200 - but as we've been discussing, the python server doesn't do the 404 anyway, so the change isn't necessary.